TST: TestQuantityReshapeFuncs.test_flat_attributes failed with AssertionError on s390x (big-endian)

Example log: https://github.com/astropy/astropy/runs/2708622607?check_suite_focus=true

Last week's run was successful (https://github.com/astropy/astropy/runs/2653579480?check_suite_focus=true) on 2021-05-24.

```
________________ TestQuantityReshapeFuncs.test_flat_attributes _________________

self = <astropy.units.tests.test_quantity_array_methods.TestQuantityReshapeFuncs object at 0x42b00b16a0>

    def test_flat_attributes(self):
        """While ``flat`` doesn't make a copy, it changes the shape."""
        q = np.arange(6.).reshape(3, 1, 2) * u.m
        qf = q.flat
        # see TestQuantityArrayCopy.test_flat for tests of iteration
        # and slicing and setting. Here we test the properties and methods to
        # match `numpy.ndarray.flatiter`
        assert qf.base is q
        # testing the indices -- flat and full -- into the array
        assert qf.coords == (0, 0, 0)  # to start
        assert qf.index == 0
        # now consume the iterator
        endindices = [(qf.index, qf.coords) for x in qf][-2]  # next() oversteps
>       assert endindices[0] == 5
E       assert 0 == 5

astropy/units/tests/test_quantity_array_methods.py:138: AssertionError
```

Could this be related to #11796 , @mhvk or @nstarman ?

Also see #11783, numpy/numpy#19153

--- a/astropy/units/tests/test_quantity_array_methods.py
+++ b/astropy/units/tests/test_quantity_array_methods.py
@@ -1,11 +1,15 @@
 # The purpose of these tests are to ensure that calling quantities using
 # array methods returns quantities with the right units, or raises exceptions.
+import os
 
 import pytest
 import numpy as np
 from numpy.testing import assert_array_equal
 
 from astropy import units as u
+from astropy.utils.compat import NUMPY_LT_1_22
+
+IS_S390X = os.environ.get('ARCH_ON_CI', 'normal') == 's390x'
 
 
 class TestQuantityArrayCopy:
@@ -122,6 +126,7 @@ class TestQuantityReshapeFuncs:
         assert q_swapaxes.unit == q.unit
         assert np.all(q_swapaxes.value == q.value.swapaxes(0, 2))
 
+    @pytest.mark.xfail(IS_S390X and NUMPY_LT_1_22, reason="Numpy GitHub Issue 19153")
     def test_flat_attributes(self):
         """While ``flat`` doesn't make a copy, it changes the shape."""
         q = np.arange(6.).reshape(3, 1, 2) * u.m

--- a/astropy/units/tests/test_quantity_array_methods.py
+++ b/astropy/units/tests/test_quantity_array_methods.py
@@ -1,6 +1,6 @@
 # The purpose of these tests are to ensure that calling quantities using
 # array methods returns quantities with the right units, or raises exceptions.
-import os
+import sys
 
 import pytest
 import numpy as np
@@ -8,8 +8,6 @@ from numpy.testing import assert_array_equal
 
 from astropy import units as u
 from astropy.utils.compat import NUMPY_LT_1_22
-
-IS_S390X = os.environ.get('ARCH_ON_CI', 'normal') == 's390x'
 
 
 class TestQuantityArrayCopy:
@@ -126,7 +124,8 @@ class TestQuantityReshapeFuncs:
         assert q_swapaxes.unit == q.unit
         assert np.all(q_swapaxes.value == q.value.swapaxes(0, 2))
 
-    @pytest.mark.xfail(IS_S390X and NUMPY_LT_1_22, reason="Numpy GitHub Issue 19153")
+    @pytest.mark.xfail(sys.byteorder == 'big' and NUMPY_LT_1_22,
+                       reason="Numpy GitHub Issue 19153")
     def test_flat_attributes(self):
         """While ``flat`` doesn't make a copy, it changes the shape."""
         q = np.arange(6.).reshape(3, 1, 2) * u.m

--- a/astropy/utils/compat/numpycompat.py
+++ b/astropy/utils/compat/numpycompat.py
@@ -5,7 +5,7 @@ earlier versions of Numpy.
 """
 from astropy.utils import minversion
 
-__all__ = ['NUMPY_LT_1_18', 'NUMPY_LT_1_19', 'NUMPY_LT_1_20']
+__all__ = ['NUMPY_LT_1_18', 'NUMPY_LT_1_19', 'NUMPY_LT_1_20', 'NUMPY_LT_1_22']
 
 # TODO: It might also be nice to have aliases to these named for specific
 # features/bugs we're checking for (ex:
@@ -13,3 +13,4 @@ __all__ = ['NUMPY_LT_1_18', 'NUMPY_LT_1_19', 'NUMPY_LT_1_20']
 NUMPY_LT_1_18 = not minversion('numpy', '1.18')
 NUMPY_LT_1_19 = not minversion('numpy', '1.19')
 NUMPY_LT_1_20 = not minversion('numpy', '1.20')
+NUMPY_LT_1_22 = not minversion('numpy', '1.22')

